pkg/state/nodepoolhealth/tracker.go

```diff
@@ -82,49 +85,73 @@ func (t *Tracker) Status() Status {
 }
 
 func (t *Tracker) SetStatus(status Status) {
+	t.Lock()
+	defer t.Unlock()
+
 	switch status {
 	case StatusUnknown:
 		t.buffer.Reset()
```

Comment:
Don't we now have a deadlock issue here since `Reset` also acquires a lock? I'd consider just adding private methods (e.g. `reset` and `update`) which don't acquire the lock and call them directly here. Then the public versions of those methods are just locked wrappers.

Reply:
`Reset` doesn't acquire a lock though.

Reply:
I was looking at `(*Tracker) Reset` and got my wires crossed, lgtm then 🚀